tracer/src/Datadog.Trace/Iast/Vulnerability.cs

```diff
@@ -36,6 +45,6 @@ public Vulnerability(string type, Location location, Evidence evidence, Integrat
     public override int GetHashCode()
     {
         // We do not calculate the hash according to the evidence
-        return IastUtils.GetHashCode(Type, Location);
+        return _fixedhash ?? IastUtils.GetHashCode(Type, Location);
```

Comment:
minor: seems a little odd to have a field to store the hash, but only use it sometimes, might be better to use the field all the time?